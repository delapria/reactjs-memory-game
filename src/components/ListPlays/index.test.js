import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ListPLays from '.';

describe('Snapshot ListPLays', () => {
  test('Testar se o componente renderiza de acordo com a snapshot', () => {
    const listPLays = [
      {
        id: '1',
        playerName: 'Douglas',
        numPlays: 13,
      },
    ];
    const treeJSON = renderer
      .create(<ListPLays listPLays={listPLays} />)
      .toJSON();
    expect(treeJSON).toMatchSnapshot();
  });
});

describe('Shallow tests', () => {
  test('Testa valor apresentado pelo componente', () => {
    const listPLays = [
      {
        id: '1',
        playerName: 'Douglas',
        numPlays: 13,
      },
    ];
    const comp = shallow(<ListPLays listPLays={listPLays} />);
    expect(
      comp
        .find('strong')
        .last()
        .props().children
    ).toBe('Jogadas');
    expect(
      comp
        .find('strong')
        .first()
        .props().children
    ).toBe('Jogadores');
  });
});
