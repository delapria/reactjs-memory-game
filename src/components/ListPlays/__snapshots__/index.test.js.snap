// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ListPLays Testar se o componente renderiza de acordo com a snapshot 1`] = `
<li
  className="sc-AykKC ipYTat"
>
  <strong>
    Jogadores
  </strong>
  <strong>
    Jogadas
  </strong>
</li>
`;
